atlan/assets/errors.go

```diff
@@ -772,6 +772,10 @@ func handleApiError(response *http.Response, originalError error) error {
 	if err := json.Unmarshal(body, &errorResponse); err == nil {
 		fmt.Println(errorResponse)
 		causes = errorResponse.Causes
+		// Check for Atlan-specific error code 1006
+		if errorResponse.ErrorID == "1006" || strings.Contains(errorResponse.Message, "Please provide the required payload") {
+			return ThrowAtlanError(originalError, PERMISSION_PASSTHROUGH, nil, "API token doesn't have necessary permissions")
+		}
```

Comment:
I think this occurs when the API token lacks `$admin` privileges to run a workflow -- so we're throwing a custom, meaningful error through the SDK, right?  🤔 

Let's make it stricter or more explicit to this specific case (i.e: `errorID && message contains`). It maybe possible that the returned `errorMessage` is something like *'Please provide the required payload'*, which could actually mean the request payload is incorrect. But since we're using an `||` check, we end up returning `'API token doesn't have necessary permissions'`, which would be misleading 🤔 

```suggestion
		// Check for Atlan-specific error code 1006
		if errorResponse.ErrorID == "1006" && strings.Contains(errorResponse.Message, "Please provide the required payload") {
			return ThrowAtlanError(originalError, PERMISSION_PASSTHROUGH, nil, "API token doesn't have necessary permissions")
		}
```

Reply:
Nice catch @Aryamanz29 ! I overlooked it